### PR TITLE
Concave particle pressure validation

### LINKED ISSUE #43: Validate expansive SDF.
## Description

Add tests that exercise the expansive SDF: https://github.com/glotzerlab/hoomd-blue/pull/1391
* Concave, interlocking particles.
* A non-monotonic potential (i.e. shifted square well).

## Proposed solution

Follow the existing test suites and provide NVT/NPT consistency checks.

## Additional context

The GPU code path for user potentials is already well tested. These tests need only run on the CPU to verify that SDF works properly.

## Developer

@tcmoore3, can you implement this before the HOOMD v4 release?

### PR DESCRIPTION
**Pull request by tcmoore3**
## Description

Performs cross validation using NVT and NPT simulations of a hard particle with a concave shape. Computes the pressure in NVT simulations, and then runs NPT simulations and calculates the density the mean pressure. 

## Motivation and context

Validates the implementation of the SDF-based pressure calculation where the expansive contribution to the pressure is nonzero (i.e., for non-convex hard particles).

Resolves #43

## How has this been tested?

Ran on PSC-Bridges2. Results:
image

## Checklist:

- [ ] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-validation/blob/trunk/CONTRIBUTING.md).
- [ ] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-validation/blob/trunk/ContributorAgreement.md).
- [ ] My name is on the list of contributors (`CONTRIBUTORS.md`) in the pull request source branch.

### CODE REVIEW

**Comment by tcmoore3** [comment]
pre-commit.ci autofix

**Comment by tcmoore3** [comment]
pre-commit.ci autofix

File: hoomd_validation/init.py [hunk 1]
```diff
@@ -13,8 +13,11 @@
 import lj_union
 import hard_disk
 import hard_sphere
+import simple_polygon
 
-subprojects = [alj_2d, lj_fluid, lj_union, hard_disk, hard_sphere]
+subprojects = [
+    alj_2d, lj_fluid, lj_union, hard_disk, hard_sphere, simple_polygon
```
Comment (joaander) [inline hoomd_validation/init.py]:
```suggestion
    alj_2d, lj_fluid, lj_union, hard_disk, hard_sphere, simple_polygon,
```
And run `pre-commit` to format one item per line.

File: hoomd_validation/simple_polygon.py [hunk 2]
```diff
@@ -0,0 +1,652 @@
+# Copyright (c) 2022-2023 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+"""Hard disk equation of state validation test."""
```
Comment (joaander) [inline hoomd_validation/simple_polygon.py]:
```suggestion
"""Simple polygon equation of state validation test."""
```

File: hoomd_validation/simple_polygon.py [hunk 3]
```diff
@@ -0,0 +1,652 @@
...
+
+def job_statepoints():
+    """list(dict): A list of statepoints for this subproject."""
+    num_particles = 64**2
+    replicate_indices = range(CONFIG["replicates"])
+    # reference statepoint from: http://dx.doi.org/10.1016/j.jcp.2013.07.023
+    # Assume the same pressure for a 128**2 system as this is in the fluid
+    # from first test: mean +- std_err_means =
+    # 6.205344838518146 +- 0.0011239315686705335
+    # based on nvt sims at density = 0.75 over 32 replicas
```
Comment (joaander) [inline hoomd_validation/simple_polygon.py]: Not all of these comment lines can be true at the same time. Remove the invalid ones.

File: hoomd_validation/simple_polygon.py [hunk 4]
```diff
@@ -0,0 +1,652 @@
...
+import os
+import json
+import pathlib
+
+# Run parameters shared between simulations.
+# Step counts must be even and a multiple of the log quantity period.
+RANDOMIZE_STEPS = 20_000
+EQUILIBRATE_STEPS = 100_000
+RUN_STEPS = 2_500_000
+RESTART_STEPS = RUN_STEPS // 10
```
Comment (joaander) [inline hoomd_validation/simple_polygon.py]:
```suggestion
RESTART_STEPS = RUN_STEPS // 50
```
Shorten the restart period from ~30 minutes down to ~6 (tested on cheme-hodges).

**Review by joaander — CHANGES_REQUESTED** [review]
Thanks! Please post the analysis results including the `nvt_gpu` jobs (run on Great Lakes or Delta).

**Comment by tcmoore3** [comment]
Here are the results from the GPU jobs run on Delta.
image

**Review by joaander — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**